src/detect-parse.c

```diff
@@ -858,6 +864,76 @@ int SigMatchListSMBelongsTo(const Signature *s, const SigMatch *key_sm)
     return -1;
 }
 
+/**
+ * \brief Parse and setup a direction
+ *
+ * \param s signature
+ * \param str argument to the keyword
+ * \param only_dir argument wether the keyword only accepts a direction
+ *
+ * \retval 0 on success, -1 on failure
+ */
+static int DetectSetupDirection(Signature *s, char **str, bool only_dir)
+{
+    if (strncmp(*str, "to_client", strlen("to_client")) == 0) {
```

Comment:
this will also match `to_client_toto` I think. Will that be detected correctly?

Reply:
yes

After matching the prefix, we check further characters : optional spaces then comma or end of string

`to_client_toto ` gets to line 887 SCLogError("expecting to_client [, other]");